etherman/etherman.go

```diff
@@ -233,6 +245,9 @@ func (etherMan *Client) WaitTxToBeMined(hash common.Hash, timeout time.Duration)
 
 // EstimateGasSequenceBatches estimates gas for sending batches
 func (etherMan *Client) EstimateGasSequenceBatches(sequences []ethmanTypes.Sequence) (*types.Transaction, error) {
+	if etherMan.readOnly {
+		return nil, errors.New("etherman client in read-only mode, cannot estimate gas for sending batches")
```

Comment:
Small suggestion, add a new error in line 48 like:

    ErrIsReadOnlyMode = errors.New("etherman client in read-only mode, cannot estimate gas for sending batches")

Then:

```suggestion
                return nil, ErrIsReadOnlyMode
````

Also for line 264.

Reply:
@KonradIT  good point, I applied it on every occurrence here ed8a793590e46a073c12307adb7b6801c883f38c